Update group deployer to support multiple routes

kaufers opened the pull request.

kaufers committed:
Update group deployer to support multiple routes

The current group deployer uses the value in the ROUTE_HOSTNAME environment
variable to specific the single hostname to map to the container group.
This commit adds support for a ADDITIONAL_HOSTNAMES environment variable
that is a comma-separated list of hostnames that should all be mapped to
the container group. All hostnames specified (the union of ROUTE_HOSTNAME
and ADDITIONAL_HOSTNAMES) will be mapped to the new group and unmapped
from any existing groups.

All hostname will share the common ROUTE_DOMAIN domain.

kaufers committed:
Merge remote-tracking branch 'upstream/master'

jgarcows commented:
The done (line 287) should be moved to above line 281.  The writing of the DEPLOY_PROPERTY_FILE should be outside of the loop but still within the if.

kaufers committed:
Move route URL addition to the deploy prop file

The deploy properties file needs to be written outside of the route loop;
it will only have the primary route (the ROUTE_HOSTNAME route) and none of
the additional ones.
